src/raftstore/errors.rs

```diff
@@ -150,6 +150,7 @@ impl Into<errorpb::Error> for Error {
                 }
                 errorpb.mut_not_leader().set_region_id(region_id);
             }
+            Error::StoreNotMatch(..) => errorpb.set_store_not_match(errorpb::StoreNotMatch::new()),
```

Comment:
Tidb will clear the expected store id's address in cache when meet this error. @siddontang